### PR TITLE
Speed up generation step

### PR DESCRIPTION
**Pull request by solomonchild**

### CODE REVIEW

File: cmake_overrides.cmake [hunk 1]
```diff
@@ -0,0 +1,1250 @@
...
+	SET(HAVE_SYS_UTSNAME_H 0)
+	SET(SIZEOF_SIZE_T 8)
+	SET(HAVE_SIZEOF_SIZE_T TRUE)
+	SET(SIZEOF_SIZE_T_CODE "#define SIZEOF_SIZE_T 8")
+	SET(SIZEOF_SSIZE_T 0)
+	SET(HAVE_SIZEOF_SSIZE_T FALSE)
+	SET(SIZEOF_SSIZE_T_CODE "#define SIZEOF_SSIZE_T 0")
+	SET(SIZEOF_LONG_LONG 8)
+	SET(HAVE_SIZEOF_LONG_LONG TRUE)
+	SET(SIZEOF_LONG_LONG_CODE "#define SIZEOF_LONG_LONG 8")
```
Comment (snake5) [inline cmake_overrides.cmake:120]: must support win32 builds too (also wrt SIZEOF_TIME_T etc. below?)

Comment (arvids-kokins-bidstack) [inline cmake_overrides.cmake]: must support win32 builds too (also possibly wrt SIZEOF_TIME_T etc.)

File: cmake_overrides.cmake [hunk 2]
```diff
@@ -0,0 +1,1250 @@
...
+	SET(HAVE_SOCKIO_H 0)
+	SET(HAVE_SYS_UTSNAME_H 1)
+	set(CURL_INCLUDES ${CURL_INCLUDES} "sys/utsname.h")
+	set(CURL_TEST_DEFINES "${CURL_TEST_DEFINES} -DHAVE_SYS_UTSNAME_H")
+	SET(SIZEOF_SIZE_T 8)
+	SET(HAVE_SIZEOF_SIZE_T TRUE)
+	SET(SIZEOF_SIZE_T_CODE "#define SIZEOF_SIZE_T 8")
+	SET(SIZEOF_SSIZE_T 8)
+	SET(HAVE_SIZEOF_SSIZE_T TRUE)
+	SET(SIZEOF_SSIZE_T_CODE "#define SIZEOF_SSIZE_T 8")
```
Comment (arvids-kokins-bidstack) [inline cmake_overrides.cmake]: appears to be repeated below

File: cmake_overrides.cmake (hunk 1 above)
Comment (solomonchild) [reply cmake_overrides.cmake]: Given that we don't provide win32 builds, what's the purpose?

File: cmake_overrides.cmake [hunk 3]
```diff
@@ -0,0 +1,1250 @@
...
+	SET(SIZEOF_SSIZE_T_CODE "#define SIZEOF_SSIZE_T 4")
+	SET(SIZEOF_LONG 4)
+	SET(SIZEOF_LONG_CODE "#define SIZEOF_LONG 4")
+	SET(SIZEOF_LONG_DOUBLE 8)
+	SET(SIZEOF_LONG_DOUBLE_CODE "#define SIZEOF_LONG_DOUBLE 8")
+	SET(SIZEOF_TIME_T 4)
+	SET(SIZEOF_TIME_T_CODE "#define SIZEOF_TIME_T 4")
+	SET(SEND_QUAL_ARG2 const)
+
+	if(${ANDROID_ABI} STREQUAL "arm64-v8a")
```
Comment (arvids-kokins-bidstack) [inline cmake_overrides.cmake]: the following can probably be combined into 2 ifs for 32/64bit architectures?

File: CMakeLists.txt [hunk 4]
```diff
@@ -1127,11 +1131,14 @@ if(HAVE_FILE_OFFSET_BITS)
 endif()
 check_type_size("off_t"  SIZEOF_OFF_T)
 
-# include this header to get the type
-set(CMAKE_REQUIRED_INCLUDES "${CURL_SOURCE_DIR}/include")
-set(CMAKE_EXTRA_INCLUDE_FILES "curl/system.h")
-check_type_size("curl_off_t"  SIZEOF_CURL_OFF_T)
-set(CMAKE_EXTRA_INCLUDE_FILES "")
+
+if(NOT "${SKIP_CURL_CHECKS}")
```
Comment (arvids-kokins-bidstack) [inline CMakeLists.txt:1135]: does it work to skip checks entirely on the first build?

Comment (solomonchild) [reply CMakeLists.txt:1135]: Yes, does it not work for you?

File: cmake_overrides.cmake (hunk 1 above)
Comment (arvids-kokins-bidstack) [reply cmake_overrides.cmake]: not completely breaking the ability to perform certain internal test builds

File: CMakeLists.txt (hunk 4 above)
Comment (arvids-kokins-bidstack) [reply CMakeLists.txt:1135]: nvm, looked like a different kind of check at first
haven't had the time to try building anything yet

File: cmake_overrides.cmake (hunk 1 above)
Comment (solomonchild) [reply cmake_overrides.cmake]: Oh, sure.

> also possibly wrt SIZEOF_TIME_T etc

It seems like `SIZEOF_SIZE_T` is the only thing that 32bit vs 64bit differ in.

**Review by arvids-kokins-bidstack — APPROVED** [review]

**Pull request merged**